src/azure-cli-core/azure/cli/core/commands/template_create.py

```diff
@@ -110,3 +122,132 @@ def handle_folding(namespace):
 
     return handle_folding
 
+# NEW STYLE
+
+# pylint: disable=line-too-long
+def get_folded_parameter_help_string(
+        display_name, allow_none=False, allow_new=False, default_none=False,
+        other_required_option=None):
+    """ Assembles a parameterized help string for folded parameters. """
+    quotes = '""' if platform.system() == 'Windows' else "''"
+
+    if default_none and not allow_none:
+        raise CLIError('Cannot use default_none=True and allow_none=False')
+
+    if not allow_new and not allow_none and not default_none:
+        help_text = 'Name or ID of an existing {}.'.format(display_name)
+    elif not allow_new and allow_none and not default_none:
+        help_text = 'Name or ID of an existing {}, or {} for none.'.format(display_name, quotes)
+    elif allow_new and not allow_none and not default_none:
```

Comment:
` and not default_none` is not needed, as the first if block has excluded this case already.